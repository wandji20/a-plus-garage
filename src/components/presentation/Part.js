import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { CircularProgressbar } from 'react-circular-progressbar';
import 'react-circular-progressbar/dist/styles.css';
import updatePartAction from '../../redux/actions/updatePartAction';

const Part = (props) => {
  const { part, handleUpdateAction } = props;
  // console.log(part);

  const handleUpdatePart = () => {
    const { id, count } = part;
    // console.log([id, count, handleUpdateAction]);
    handleUpdateAction(part.car_id, id, { count: count + 1 });
  };
  const { name, url, stats } = part;
  return (
    <article className="part my-3 align-items-center row bg-white mx-1 py-2 ">
      <figure className=" col-4 d-flex flex-column">
        <h3>
          {name}
        </h3>
        <img className="img" src={url} alt={part.name} />
      </figure>
      <div className="col-8 d-flex justify-content-between align-items-center">
        <div className="d-flex col-8 flex-column justify-content-center align-items-start">
          <h6>
            {stats.status}
            {/* {' '}
            Condition */}
          </h6>
          <p className="">
            <span className="d-block">
              {stats.daysLeft}
              {' '}
              days left!
            </span>
            <button
              type="submit"
              className="btn btn-info"
              onClick={handleUpdatePart}
            >
              Track
            </button>
          </p>

        </div>
        <div
          style={{ width: '64px', height: '64px' }}
          className="d-flex col-4 flex-column justify-content-end align-items-start"
        >
          <CircularProgressbar
            value={stats.percentage}
            text={`${stats.percentage}%`}
          />
        </div>
      </div>
    </article>
  );
};

const mapDispatchToProps = (dispatch) => ({
  handleUpdateAction: (carId, id, data) => {
    dispatch(updatePartAction(carId, id, data));
  },
});

Part.propTypes = {
  part: PropTypes.objectOf(PropTypes.any).isRequired,
  handleUpdateAction: PropTypes.func.isRequired,
};

export default connect(null, mapDispatchToProps)(Part);
