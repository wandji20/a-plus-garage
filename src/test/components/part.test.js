import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../redux/store';
import Part from '../../components/presentation/Part';
import { BrowserRouter } from 'react-router-dom';

it('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
        <BrowserRouter>
          <Part part={{name:'', life:'', stats:{}}} />
        </BrowserRouter>
      </Provider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
