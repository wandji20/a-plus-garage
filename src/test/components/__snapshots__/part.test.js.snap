// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article
  className="part my-3 align-items-center row bg-white mx-1 py-2 "
>
  <figure
    className=" col-4 d-flex flex-column"
  >
    <h3>
      
    </h3>
    <img
      alt=""
      className="img"
    />
  </figure>
  <div
    className="col-8 d-flex justify-content-between align-items-center"
  >
    <div
      className="d-flex col-8 flex-column justify-content-center align-items-start"
    >
      <h6 />
      <p
        className=""
      >
        <span
          className="d-block"
        >
           
          days left!
        </span>
        <button
          className="btn btn-info"
          onClick={[Function]}
          type="submit"
        >
          Track
        </button>
      </p>
    </div>
    <div
      className="d-flex col-4 flex-column justify-content-end align-items-start"
      style={
        Object {
          "height": "64px",
          "width": "64px",
        }
      }
    >
      <svg
        className="CircularProgressbar "
        data-test-id="CircularProgressbar"
        style={Object {}}
        viewBox="0 0 100 100"
      >
        <path
          className="CircularProgressbar-trail"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fillOpacity={0}
          strokeWidth={8}
          style={
            Object {
              "strokeDasharray": "289.02652413026095px 289.02652413026095px",
              "strokeDashoffset": "0px",
            }
          }
        />
        <path
          className="CircularProgressbar-path"
          d="
      M 50,50
      m 0,-46
      a 46,46 0 1 1 0,92
      a 46,46 0 1 1 0,-92
    "
          fillOpacity={0}
          strokeWidth={8}
          style={
            Object {
              "strokeDasharray": "289.02652413026095px 289.02652413026095px",
              "strokeDashoffset": "NaNpx",
            }
          }
        />
        <text
          className="CircularProgressbar-text"
          style={Object {}}
          x={50}
          y={50}
        >
          undefined%
        </text>
      </svg>
    </div>
  </div>
</article>
`;
